clang/lib/DPCT/SaveNewFiles.cpp

```diff
@@ -211,6 +228,19 @@ void processallOptionAction(clang::tooling::UnifiedPath &InRoot,
     if (!rewriteDir(OutputFile, InRoot, OutRoot)) {
       continue;
     }
+    // Check for another file with SYCL extension. For example
+    // In in-root we have
+    //  * src.cpp
+    //  * src.cu
+    // After migration we will end up replacing src.cpp with migrated src.cu
+    // when the --sycl-file-extension is cpp
+    // In such a case warn the user.
+
+    // Make sure that the output file corresponds to a single and unique input
+    // file.
+    if (checkOverwriteAndWarn(OutputFile.getCanonicalPath().str(), File))
```

Comment:
If the type of `checkOverwriteAndWarn`'s 1st and 2nd parameter is StringRef type, we can avoid create a materialized std::string temporary here.

Reply:
Make sense. Done